src/program.cpp

```diff
@@ -111,342 +110,115 @@ program& program::operator=(program p)
 
 void program::assign(const program& p)
 {
-    // clean the current program
```

Comment:
Cant this be implemented as `*this = p;`?

Reply:
*this = p is going to be a shallow copy? the two program instances will share the instructions. I think we need a deep copy here.

Reply:
I just tested it, and *this = p does work for many our unit tests.

Reply:
The assignment operator should make a copy of the program.